tests/rptest/remote_scripts/stream_verifier_txn.py

```diff
@@ -0,0 +1,974 @@
+    """
+    def __init__(self,
+                 brokers,
+                 topics_config: dict,
+                 worker_threads: int,
+                 rate=0,
+                 total_messages=100):
+        # Remove quotes if any
+        self.logger = setup_logger('core')
+        # Remove quotes from broker config value if an
+        self.brokers = brokers.strip('\"').strip("'")
+        # Create main topics config
+        self.topics_config = TopicsConfig(**topics_config)
+        self.message_rate = rate
+        # It is reasonable to assume that single message will not be sent
+        # faster than 1 ms in case of no rate limitations
+        self.msgs_rate_ms = 1000 / self.message_rate if rate > 0 else 0
+        # total messages to process in all topics
+        self.total_messages = total_messages
+        self.workers = worker_threads
+        # Annoucement of dynamic vars
+        self.topics_status = {}
+        self.topics = {}
+        self.id_index = 0
+        self.produce_thread = None
+        self.consume_thread = None
+        self.delivery_reports = {}
+
+    def init_producers(self):
+        """Precreates topic status lists and initializes Producers.
+        One producer per topic.
+        """
+        self.logger.info("Initializing producers")
+        # Calculate messages per topic and announce changes
+        msgs_per_topic = int(self.total_messages /
+                             self.topics_config.topic_count)
+        new_total_messages = msgs_per_topic * self.topics_config.topic_count
+        if new_total_messages != self.total_messages:
+            self.logger.warning("Messages per topic rounded to "
+                                f"{msgs_per_topic} with the new "
+                                f"total of {new_total_messages}")
+            self.total_messages = new_total_messages
+        else:
+            self.logger.info(f"Messages per topic is {msgs_per_topic} "
+                             f"with the total of {self.total_messages}")
+        for name in self.topics_config.names:
+            topic_config = {
+                "id": self.id_index,
+                "name": name,
+                "msgs_per_transaction": app_config.msg_per_txn,
+                "total_messages": msgs_per_topic,
+                "index": 0,
+                "last_message_ts": datetime.now().timestamp(),
+                "msgs_rate_ms": self.msgs_rate_ms,
+                "consume_timeout": app_config.consume_timeout,
+                "producer": None,
+                "consumer_config": {},
+                "terminate": False
+            }
+            t = TopicStatus(**topic_config)
+            t.producer = ck.Producer({
+                'bootstrap.servers': self.brokers,
+                'transactional.id': t.transaction_id
+            })
+            t.producer.init_transactions()
+            self.topics[t.name] = t
+
+    def _send_transaction(self, logger: logging.Logger,
```

Comment:
nit: the usage of "_" is inconsistent, some class methods have it some don't.. I'd vote for getting rid of them totally, we don't use that pattern widely in ducktape tests.